### DISCUSSION TITLE
svgWithLogo.php example not running

### DISCUSSION THREAD

**Discussion by sogorman** [chillerlan/php-qrcode#135] [General]
Appologies for what is probably a noob question...

I am trying to get the https://github.com/chillerlan/php-qrcode/blob/main/examples/svgWithLogo.php example running.

 I installed the latest version by running `composer require chillerlan/php-qrcode` and then pulled down the svgLogo example php file into that project's example directory. I updated the require to be `require_once '/var/www/html/qr/vendor/autoload.php';` but when running that php file i receive these exceptions

`[Wed Jul 06 20:59:16.061095 2022] [php:error] [pid 19119] [client 10.0.0.170:53094] PHP Fatal error:  Uncaught Error: Class "chillerlan\\QRCodeExamples\\EccLevel" not found in /var/www/html/qr/vendor/chillerlan/php-qrcode/examples/svg.php:31\nStack trace:\n#0 {main}\n  thrown in /var/www/html/qr/vendor/chillerlan/php-qrcode/examples/svg.php on line 31

[Wed Jul 06 21:03:20.949402 2022] [php:error] [pid 19117] [client 10.0.0.170:53136] PHP Fatal error:  Uncaught Error: Undefined constant chillerlan\\QRCode\\Data\\QRMatrix::IS_DARK in /var/www/html/qr/vendor/chillerlan/php-qrcode/examples/svgWithLogo.php:48\nStack trace:\n#0 {main}\n  thrown in /var/www/html/qr/vendor/chillerlan/php-qrcode/examples/svgWithLogo.php on line 48
` 

When looking at the files in the project it looks like there are files in the repo in say SRC that are not in the composer install.

**Comment by codemasher** [comment]
Looks like you got a version mismatch between your installed version and the example from the `main` (development) branch. You can try either [installing the dev version](https://stackoverflow.com/a/41703068) by running `composer require chillerlan/php-qrcode:dev-main#2777a0d9125b9f214495ee63e008386942d3f5b0` or use [the examples from the 4.3.x branch over here](https://github.com/chillerlan/php-qrcode/tree/v4.3.x).

**Reply by sogorman** [reply]
@codemasher  yea, you are correct, that resolved it. Apologies

**Reply by codemasher** [reply]
no need to apologize! It's on me - i should release v5 at some point to avoid further confusion, but it still needs a bit polish.

**Reply by sogorman** [reply]
May I send you a PM / DM. Is a good contact s****y@c*******n.net?

**Reply by mshannaq** [reply]
@codemasher  Yes, I've noticed that when I was using the command `composer require chillerlan/php-qrcode`, several features were unavailable. However, when I switched to using `chillerlan/php-qrcode:dev-main`, the features described in the documentation worked perfectly. 

By the way, I wanted to point out that the URL https://php-qrcode.readthedocs.io/en/main/Usage-Advanced-usage.html, which is provided in the https://github.com/chillerlan/php-qrcode/blob/main/README.md for advanced usage, appears to be a broken link.

**Reply by codemasher** [reply]
Hi @mshannaq! There's a 5.0-beta release already you might want to check out. However, it still may differ a bit from dev-main at this point.
